packages/storefront-lwc/src/modules/commerce/productDetail/productDetail.js

```diff
@@ -143,34 +146,25 @@ export default class ProductDetail extends LightningElement {
     handleCarousel(event) {
         const { slide } = event.currentTarget.dataset;
         if (slide === 'prev') {
-            this.setActiveImageCss(
+            this.activeImage =
                 this.activeImage === 0
                     ? this.product.images.length - 1
-                    : this.activeImage - 1,
-            );
+                    : this.activeImage - 1;
         } else {
-            this.setActiveImageCss(
+            this.activeImage =
                 this.activeImage === this.product.images.length - 1
                     ? 0
-                    : this.activeImage + 1,
-            );
+                    : this.activeImage + 1;
         }
     }
 
-    /**
-     * Set the active image for the product detail carousel
-     * @param activeImage the url of the image to be displayed
-     */
-    setActiveImageCss(activeImage) {
-        this.product.cssClass = 'carousel-item';
-        this.activeImage = activeImage;
-        if (this.product && this.product.images) {
-            this.product.images.forEach((image, idx) => {
-                image.cssClass =
-                    idx === this.activeImage
-                        ? 'carousel-item active'
-                        : 'carousel-item';
-            });
-        }
+    get images() {
```

Comment:
Use a getter with a map for the images instead of directly mutating the backend response